crates/ruff/src/rules/pep8_naming/rules/invalid_module_name.rs

```diff
@@ -72,3 +79,21 @@ pub fn invalid_module_name(path: &Path, package: Option<&Path>) -> Option<Diagno
 
     None
 }
+
+/// Return `true` if a [`Path`] should use the name of its parent directory as its module name.
+fn is_module_file(path: &Path) -> bool {
+    path.file_name().map_or(false, |file_name| {
+        file_name == "__init__.py"
+            || file_name == "__init__.pyi"
+            || file_name == "__main__.py"
+            || file_name == "__main__.pyi"
+    })
+}
+
+/// Return `true` if a [`Path`] refers to a migration file.
+fn is_migration_file(path: &Path) -> bool {
+    path.parent()
+        .and_then(Path::file_name)
+        .and_then(OsStr::to_str)
+        .map_or(false, |parent| matches!(parent, "versions" | "migrations"))
+}
```

Comment:
(E.g., to me, this check is similar to the `is_module_file` check we already do here -- it feels more at-home in the rule than in the compliance utilities...)